File: 프로그래머스/LV2/구명보트.py
from collections import deque


def solution(people, limit):
    answer = 0
    people = deque(sorted(people))

    while people:
        person = people.pop()

        if len(people) > 0 and people[0] + person <= limit:
            people.popleft()
        answer += 1

    return answer

File: 프로그래머스/javascript/LV2/구명보트.js
function solution(people, limit) {
    people.sort((a, b) => a - b);
    let left = 0;
    let right = people.length - 1;
    let boats = 0;

    while (left <= right) {
        // 가장 가벼운 사람 + 가장 무거운 사람이 같이 탈 수 있으면 함께
        if (people[left] + people[right] <= limit) {
            left++;
        }
        // 무거운 사람은 항상 보트 탑승
        right--;
        boats++;
    }

    return boats;
}